src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -0,0 +1,154 @@
+package de.tum.in.www1.artemis.service;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.time.ZonedDateTime;
+import java.util.*;
+
+import org.apache.commons.fileupload.*;
+import org.apache.commons.fileupload.disk.DiskFileItem;
+import org.apache.pdfbox.multipdf.Splitter;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDDocumentInformation;
+import org.apache.pdfbox.text.PDFTextStripper;
+import org.apache.tomcat.util.http.fileupload.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import de.tum.in.www1.artemis.domain.Attachment;
+import de.tum.in.www1.artemis.domain.enumeration.AttachmentType;
+import de.tum.in.www1.artemis.domain.lecture.AttachmentUnit;
+import de.tum.in.www1.artemis.service.util.Tuple;
+import de.tum.in.www1.artemis.web.rest.dto.LectureUnitSplitDTO;
+import de.tum.in.www1.artemis.web.rest.dto.LectureUnitsDTO;
+
+@Service
+public class LectureUnitProcessingService {
+
+    private final Logger log = LoggerFactory.getLogger(FileService.class);
+
+    public List<LectureUnitsDTO> splitUnits(List<LectureUnitSplitDTO> lectureUnitSplitDTOs, MultipartFile file) throws IOException {
```

Comment:
I would recommend addng javadoc and comments to this method and to the one below. It is a littele bit hard to understand imo what is happening here

Reply:
@stefanwaldhauser can you check the explanations (JavaDoc)?